### PR TITLE
source/mode: remove mode suffix from modes

### PR DESCRIPTION
**Pull request by jmercouris**
it is duplicative to both have them in a mode folder, AND have the
mode suffix

### CODE REVIEW

**Comment by jmercouris** [comment]
Additionally, rename `application-mode` to `passthrough-mode` that is what most people seem to know it as.

**Comment by Ambrevar** [comment]
I'm confused: you've already pushed https://github.com/atlas-engineer/nyxt/commit/779749f3f23fafdfc34aae7d73fa639315161a5d.
Was it a mistake?

We should avoid too many renames, it makes it hard to follow through the Git history later on.

Otherwise I agree with this change :)
